src/components/Image/Image.st.css

```diff
@@ -1,2 +1,32 @@
+:vars {
+  BlurRadius: 3px;
+}
+
+@keyframes blurry-loading {
+  0% {
+    filter: blur(value(BlurRadius));
+  }
+  100% {
+    filter: none;
+  }
+}
+
 .root {
+  -st-states: preload, loaded;
+
+  display: grid;
+}
+
+.root img {
+  grid-area: 1 / 1;
+}
+
+.root:preload img:last-child {
+  filter: blur(value(BlurRadius));
+}
+
+.root:loaded img:last-child {
+  animation-name: blurry-loading;
+  animation-duration: 0.3s;
+  animation-timing-function: easeout;
 }
```

Comment:
After adding the `className` prop to the MediaImage component, I would give different class names to the two images, just so not to rely on the dom ordering, and for better readability.